.github/workflows/dbld-images.yml

```diff
@@ -35,15 +29,15 @@ jobs:
           gh_export CONTAINER_REGISTRY
 
       - name: Build the images
-        run: dbld/rules images -j $(nproc)
+        run: dbld/rules images -j $(nproc) --keep-going --output-sync
 
       - name: Should we upload the images?
+        if: always()
         run: |
-          if [ "${{ github.event.inputs.upload_images }}" = "true" ] || \
+          if [ "${{ github.event.inputs.testing_image_upload }}" = "true" ] || \
              ( \
               [ "${{ github.repository_owner }}" = "syslog-ng" ] && \
-              [ "${{ github.ref }}" = "refs/heads/master" ] && \
-              [ "${{ github.event_name }}" = "push" ] \
+              [ "${{ github.ref }}" = "refs/heads/master" ] \
```

Comment:
I understand that there is filtering in the `on:` keyword above, so we do not have to check them here, but I am afraid that someone adds a new `on:` trigger, without thinking through the effects of it in this step. Maybe we should be stricter here. What do you think?

Reply:
I see your point, but I wouldn't add that extra protection: right now every trigger that is listed under the `on` keyword is allowed to upload images.
I would only start explicitly checking for the event type if we introduce events that are not allowed to upload images.

Reply:
Why is `pull_request` allowed to upload images? It will upload to the fork's registry, and the images will never be used later, so IMO it is unnecessary.

Reply:
Well, it's not true what I said: we run the github actions on pull requests and it's not allowed to upload images from a pull request.
My bad, sorry.

I'll add an explicit check for event types: `workflow_dispatch`, `push` and `schedule` will be allowed to upload images.
What do you think?

Reply:
I like that idea :)

Reply:
I've added a check for the github event type using bash pattern matching with regex (`=~`).
I've tested the logic with a modified yaml file in my fork:
https://github.com/gaborznagy/syslog-ng/runs/4584715528?check_suite_focus=true